rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -54,6 +55,24 @@ struct SubscriptionOptionsBase
   /// Optional RMW implementation specific payload to be used during creation of the subscription.
   std::shared_ptr<rclcpp::detail::RMWImplementationSpecificSubscriptionPayload>
   rmw_implementation_payload = nullptr;
+
+  // Options to configure topic statistics collector in the subscription.
+  struct TopicStatisticsOptions
+  {
+    // Represent the state of topic statistics collector.
+    enum class TopicStatisticsState {ENABLED, DISABLED};
+
+    // Enable and disable topic statistics calculation and publication. Defaults to disabled.
+    TopicStatisticsState state = TopicStatisticsState::DISABLED;
+
+    // Topic to which topic statistics get published when enabled. Defaults to /statistics.
+    std::string publish_topic = "/statistics";
+
+    // Topic statistics publication period in ms. Defaults to one minute.
+    std::chrono::milliseconds publish_period{std::chrono::minutes(1)};
```

Comment:
Just out of curiosity is this the window used to calculate statistics or how often it publishes an update? A minute seems like a long period if it is the latter.

Reply:
It is the latter - how often the collected metrics are published. It is set to a big value so the statistics messages don't spam the network by default. Would one second be more appropriate?

Reply:
Again, without seeing how it is used it's hard to say, but if I imagine a tool that displays this information live or something, minute to minute is probably not frequent enough, imo.

Reply:
One second could be good. I don't think that's too bad.

Reply:
This is another situation where changing all of the subscriptions at once, globally or at a node level, would be nice to be able to do without changing the arguments to the constructor of the subscription directly.

Reply:
Changed default to one second.

I have changed the `topic_statistics_options.state` to have NodeDefault like `use_intra_process` does. I'm not sure how to do the same for `publish_period`.